src/test/java/br/com/caelum/clines/api/promotionalcodes/PromotionalCodeControllerTest.java

```diff
@@ -0,0 +1,61 @@
+package br.com.caelum.clines.api.promotionalcodes;
+
+import br.com.caelum.clines.shared.domain.PromotionalCode;
+import com.google.gson.Gson;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.AutoConfigureTestEntityManager;
+import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.web.servlet.MockMvc;
+
+import javax.transaction.Transactional;
+
+import java.util.Calendar;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.header;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@TestPropertySource(properties = {"DB_NAME=clines_test", "spring.jpa.hibernate.ddlAuto:create-drop"})
+@AutoConfigureMockMvc
+@AutoConfigureTestEntityManager
+@Transactional
+public class PromotionalCodeControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private TestEntityManager entityManager;
+
+    private Gson gson = new Gson();
```

Comment:
This field is not being used on any test. I think it could be removed.

Reply:
Done

Reply:
Fixed